test/unit/test_logger.py

```diff
@@ -126,3 +126,28 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors_0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors_1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+def test_markup_detection_is_tty(monkeypatch, mocker):
```

Comment:
mocker and monkeypatch fixtures have the same purpose. I suggest you go for built-in monkeypatch.